### ISSUE TITLE
[BUG] example code to insert source doc as attachment in custom step doesn't work with XML

### ISSUE THREAD

**Issue by grtjn** [marklogic/marklogic-data-hub#3282]
We created a new datahub project using quickstart 5.0.2, created an entity with a flow, and an ingest step for it to load a csv as XML. We then proceeded by appending a custom step, and the only change we make in its code is to loop through the source doc as attachment by uncommenting the following line in main.sjs:

      // instance['$attachments'] = doc;

The step fails with a DOM exception however, which is no surprise, as `instance` contains an XML Node, and no JSON.

It also seems as if makeEnvelope does not provide any other way of passing through an attachment. Looking through the source code, it seems attachments for XML envelopes are not supported, although it looks like a mistake rather than a feature.

As workaround we convert the instance to JSON, so that we can use the line as intended, and have makeEnvelope do what we expect, using:

    let jsonInstance = datahub.flow.flowUtils.xmlToJson(instance).instance;
    instance = jsonInstance[jsonInstance.info.title];
    instance['$type'] = jsonInstance.info.title;
    instance['$version'] = jsonInstance.info.version;
    instance['$attachments'] = doc;

or as last line:

    instance['$attachments'] = doc.xpath('/*:envelope/*:attachments/node()').toArray()

**Comment by grtjn** [comment]
Other alternative is to meddle with the envelope after creation, with something like:

```
let newEnv = new NodeBuilder()
  .startDocument()
    .startElement(env.root.nodeName, env.root.namespaceURI);

env.xpath('/*:envelope/(*:headers|*:triples|*:instance)').toArray().forEach(child => {
  newEnv.addNode(child);
});

newEnv.startElement('attachments', env.root.namespaceURI)
  .addNode(instance.firstChild);
  .endElement();

newEnv.endElement()
  .endDocument();
```

**Comment by grtjn** [comment]
I also had to add a .xpath to prevent headers from getting wrapped with each step:

```
  //gets headers, return null if cannot be found
  let headers = datahub.flow.flowUtils.getHeaders(doc).xpath('node()') || {};
```

**Comment by wooldridge** [comment]
@grtjn, thanks. @aebadirad, is this a bug?

**Comment by taatuut** [comment]
I'd say this a bug because the default DHF code provided does not work with XML ootb.

**Comment by sbayatpur** [comment]
Jira ticket for this: https://project.marklogic.com/jira/browse/DHFPROD-4076

**Comment by kkanthet** [comment]
Issue Fixed in 5.4.0 release
